form/src/main/java/org/dhis2/form/ui/provider/inputfield/FieldProvider.kt

```diff
@@ -263,29 +265,41 @@ private fun ProvideInputText(
     modifier: Modifier = Modifier,
     fieldUiModel: FieldUiModel,
     intentHandler: (FormIntent) -> Unit,
+    uiEventHandler: (RecyclerViewUiEvents) -> Unit,
 ) {
-    var value by remember {
-        mutableStateOf(fieldUiModel.value)
-    }
-    InputText(
-        modifier = modifier.fillMaxWidth(),
-        title = fieldUiModel.label,
-        state = fieldUiModel.inputState(),
-        supportingText = fieldUiModel.supportingText(),
-        legendData = fieldUiModel.legend(),
-        inputText = value ?: "",
-        isRequiredField = fieldUiModel.mandatory,
-        onValueChanged = {
-            value = it
-            intentHandler(
-                FormIntent.OnTextChange(
-                    fieldUiModel.uid,
-                    value,
-                    fieldUiModel.valueType,
-                ),
+    when (fieldUiModel.renderingType) {
+        UiRenderType.QR_CODE -> {
+            ProvideQRInput(
```

Comment:
I would not call a provide from another provide, maybe you can check this in line 77, if ValueType is Text, then you check the render type and call provideText or provideQR, what do you think?

Reply:
That's how I initially set it up but the method got flagged by Sonar for having too much complexity if I managed it there